Anura cannot be compiled with bundled boost library

Trying to build Anura b70cc28c91de2f7b1492b3df3549bda1017b65f4
There is a compilation error:

```
Building: src/formula_callable_visitor.cpp
Linking : anura
./build/filesystem.o: In function `boost::filesystem::copy_file(boost::filesystem::path const&, boost::filesystem::path const&, boost::filesystem::copy_option::enum_type)':
/home/kroartem/SVN/frogatto/external/include/boost/filesystem/operations.hpp:492: undefined reference to `boost::filesystem::detail::copy_file(boost::filesystem::path const&, boost::filesystem::path const&, boost::filesystem::copy_option::enum_type, boost::system::error_code*)'
collect2: error: ld returned 1 exit status
Makefile:158: ошибка выполнения рецепта для цели «anura»
make: *** [anura] Ошибка 1
```

Here are my notes from Frogatto's IRC:

> <KroArtem> you seem to be using boost included in your repo, right?
> <KroArtem> seems like your boost is a bit outdated
> <KroArtem> here: https://github.com/boostorg/filesystem/blob/develop/include/boost/filesystem/operations.hpp
> <KroArtem> line 539, there is some difference with your bundled function declaration
> <KroArtem> also I found this: https://svn.boost.org/trac/boost/ticket/10038
> <KroArtem> probably your bundled boost is really outdated?

Unfortunately I didn't find how to use system-wide libboost-filesystem.

--- a/Makefile
+++ b/Makefile
@@ -89,7 +89,7 @@ LDFLAGS += -fsanitize=address
 endif
 
 # Compiler include options, used after CXXFLAGS and CPPFLAGS.
-INC := -isystem external/include $(shell pkg-config --cflags x11 sdl2 glew SDL2_image SDL2_ttf libpng zlib freetype2 cairo)
+INC := -isystem external/header-only-libs $(shell pkg-config --cflags x11 sdl2 glew SDL2_image SDL2_ttf libpng zlib freetype2 cairo)
 
 ifdef STEAM_RUNTIME_ROOT
 	INC += -isystem $(STEAM_RUNTIME_ROOT)/include

--- a/Makefile
+++ b/Makefile
@@ -138,16 +138,11 @@ INCLUDES  := $(addprefix -I,$(SRC_DIR))
 
 vpath %.cpp $(SRC_DIR)
 
+CPPFLAGS += -MMD -MP
 define cc-command
 $1/%.o: %.cpp
 	@echo "Building:" $$<
 	@$(CCACHE) $(CXX) $(BASE_CXXFLAGS) $(CXXFLAGS) $(CPPFLAGS) $(INC) $(INCLUDES) -c -o $$@ $$<
-	@$(CXX) $(BASE_CXXFLAGS) $(CXXFLAGS) $(CPPFLAGS) $(INC) $(INCLUDES) -MM $$< > $$@.d
-	@mv -f $$@.d $$@.d.tmp
-	@sed -e 's|.*:|$$@:|' < $$@.d.tmp > $$@.d
-	@sed -e 's/.*://' -e 's/\\$$$$//' < $$@.d.tmp | fmt -1 | \
-		sed -e 's/^ *//' -e 's/$$$$/:/' >> $$@.d
-	@rm -f $$@.d.tmp
 endef
 
 .PHONY: all checkdirs clean
@@ -181,4 +176,3 @@ $(foreach bdir,$(BUILD_DIR),$(eval $(call cc-command,$(bdir))))
 
 # pull in dependency info for *existing* .o files
 -include $(OBJ:.o=.o.d)
-

--- a/external/header-only-libs/glm
+++ b/external/header-only-libs/glm
@@ -1,0 +1,1 @@
+../include/glm

--- a/external/header-only-libs/treetree
+++ b/external/header-only-libs/treetree
@@ -1,0 +1,1 @@
+../include/treetree

